src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -161,6 +161,12 @@ public static class WebScriptHostBuilderExtension
                     // Logging and diagnostics
                     services.AddSingleton<IMetricsLogger>(a => new NonDisposableMetricsLogger(metricsLogger));
                     services.AddSingleton<IEventCollectorProvider, FunctionInstanceLogCollectorProvider>();
+                    if (!FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableDiagnosticEventLogging))
```

Comment:
Why did this move?

Reply:
Host would not start due to this error:

> A host error has occurred during startup operation '06dce65b-84c5-4978-96ad-09bec9ee7edc'. System.InvalidOperationException: Unable to resolve service for type 'Microsoft.Azure.WebJobs.Hosting.IPrimaryHostStateProvider' while attempting to activate 'Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.DiagnosticEventTableStorageRepository'.

I believe it's to do with the
DiagnosticEventTableStorageRepository
being registered at Web vs Script host and the primary host provider not being available at that time; I moved it and now this issue is fixed. Is there something to be concerned about or another way to address the error?

Reply:
One side effect you'll see is that we'll no longer have the logger provider injected at the Web Application level. This could impact any flows that currently raise a diagnostic event outside of the ScriptHost scope (I haven't reviewed to validate whether we have any).

This will also have a (minor) perf impact at cold start.

A safer option here might be to take a dependency on the `WebJobsScriptHostService` (which will be available at the Web Application level) and get the `IPrimaryHostStateProvider` service from the active host (or directly from `WebJobsScriptHostService`). 

That would enable you to keep the scopes/levels unchanged.

Reply:
Reverted this change and switching to getting the `IPrimaryHostStateProvider` via `_serviceProvider = scriptHostManager as IServiceProvider;`

Reply:
Coming in late to this -- but Fabio is correct that we added DiagnosticEvents at the WebHost level at one point, which means those would stop firing without these services registered there.